tech_reports/tensor_layouts/tensor_layouts.md

```diff
@@ -1,22 +1,94 @@
 # Tensor Layouts
 
 ## Introduction
+Tensors are stored in 2D memory space, referred to as a `buffer`. An N-dimensional tensor is represented as a 2D memory object by squeezing the upper dimensions into a single dimension, and the inner dimension is equivalent to the second dimension.
```

Comment:
upper dimension
vs 
inner

Should it be outer vs inner?

Also, who do we target with docs? Do we assume read is clear what inner/upper means?